include/rocksdb/advanced_options.h

```diff
@@ -1120,11 +1120,29 @@ struct AdvancedColumnFamilyOptions {
   //
   // When it's false, the user-defined timestamps will be removed from the user
   // keys when data is flushed from memtables to SST files. Other places that
-  // user keys can be persisted like WAL and blob files go through a similar
-  // process. Users should call `DB::IncreaseFullHistoryTsLow` to set a cutoff
-  // timestamp. RocksDB refrains from flushing a memtable with data still above
-  // the cutoff timestamp with best effort. When users try to read below the
-  // cutoff timestamp, an error will be returned.
+  // user keys can be persisted like file boundaries in file metadata and blob
+  // files go through a similar process. There are two major motivations
+  // for this flag:
+  // 1) backward compatibility: if the user later decides to
+  // disable the user-defined timestamp feature for the column family, these SST
+  // files can be handled by a user comparator that is not aware of user-defined
+  // timestamps.
+  // 2) enable user-defined timestamp feature for an existing column family
+  // while set this flag to be `false`: user keys in the newly generated SST
+  // files are of the same format as the existing SST files.
+  //
+  // When setting this flag to `false`, users should also call
+  // `DB::IncreaseFullHistoryTsLow` to set a cutoff timestamp for flush. RocksDB
+  // refrains from flushing a memtable with data still above
+  // the cutoff timestamp with best effort. Users can do user-defined
+  // multi-versioned read above the cutoff timestamp. When users try to read
+  // below the cutoff timestamp, an error will be returned.
+  //
+  // Note that if WAL is enabled, unlike SST files, user-defined timestamps are
+  // persisted to WAL even if this flag is set to `false`. The benefit of this
+  // is that user-defined timestamps can be recovered with the caveat that users
```

Comment:
"timestamps can be recovered"  All I see in the test is WAL being flushed to SST on recovery which gets rid of them. Did I miss a test of this?

Reply:
That's a good question. This test `DBWALTestWithTimestamp.RecoverAndFlush` intentionally set a very small `write_buffer_size` so it can trigger a flush for a memtable recovered from WAL on the recovery path, in this case, the timestamps indeed are removed because of the flush.

The other two existing tests `DBWALTestWithTimestamp.Recover`, `DBWALTestWithTimestamp.RecoverInconsistentTimestamp` reopens the DB without resetting the `write_buffer_size` to be very small, this is the majority of the actual use cases. In these cases, timestamps are recovered from WAL to the memtables and the memtables recovered from WAL logs will not be flushed until later on when it gets full.

Reply:
OK, but I don't see a case in those tests that check that timestamps are recovered from WAL, just that entries with some timestamp are recovered.

Reply:
I see what you mean, I will add this coverage in a follow up PR.